flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/AggregateUtil.scala

```diff
@@ -501,9 +497,17 @@ object AggregateUtil extends Enumeration {
       hasStateBackedDataViews: Boolean,
       needsRetraction: Boolean): AggregateInfo =
     call.getAggregation match {
+      // In the new function stack, for imperativeFunction, the conversion from
+      // BuiltInFunctionDefinition to SqlAggFunction is unnecessary, we can simply create
+      // AggregateInfo through BuiltInFunctionDefinition and runtime implementation (obtained from
+      // AggFunctionFactory) directly.
+      // NOTE: make sure to use .runtimeProvided() in BuiltInFunctionDefinition in this case.
```

Comment:
What'll happen if not use `.runtimeProvided()`?

Reply:
Should add a check if an error occurs.

Reply:
if `.runtimeProvided()` and `.runtimeClass()` both unused and without a conversion rule, the automatic conversion from `BuiltInFunctionDefinition` to `BridgingSqlAggFunction` will fail because implementation not found, and eventually throw an error, this checking logic is in `FunctionCatalogOperatorTable#verifyFunctionKind()`.

For those agg functions with a conversion rule, this is ok because even though the above process fails, they can still create `SqlAggFunction` through the conversion rule. Also, these functions are supposed to use `.runtimeDeferred()` as a mark of using conversion rule, but only a few them actually did that.

BTW, this implementation checking will not be conducted in Table API which may be a potential problem. In fact, if you delete `.runtimeProvided()` in definition, calling function in Table API can still work while SQL can't.

Reply:
So if someone adds a new agg func that is declared in the wrong way, it will not pass the test (the test case should cover both sql and table api).
In that case I have no problem.